.github/copilot-instructions.md

```diff
@@ -0,0 +1,64 @@
+# Copilot Instructions for Awesome Emoji Picker
+
+This project is a cross-browser WebExtension (Firefox, Chromium, Thunderbird) for emoji picking, clipboard, and page insertion. It is modular, uses modern JS, and is structured for maintainability and cross-platform support.
+
+## Architecture & Key Patterns
+- **Modular Structure:**
+  - `src/` contains all extension code, split by context: `background/`, `content_scripts/`, `popup/`, `options/`, `common/`.
+  - Shared logic is in `src/common/modules/` (e.g., `BrowserCommunication`, `AddonSettings`, `Logger`).
+- **Browser Communication:**
+  - Use `BrowserCommunication.addListener(COMMUNICATION_MESSAGE_TYPE.X, callback)` for cross-context messaging.
+  - Message types are defined in `src/common/modules/data/BrowserCommunicationTypes.js` as an object (used like an enum). Always use the exported constants, not raw strings.
+- **Settings:**
+  - Settings are managed via `AddonSettings` in `src/common/modules/AddonSettings/`.
+- **Localization:**
+  - All user-facing strings are localized using `_locales/` and the `Localizer` module (which uses the normal WebExtension function behind).
+- **UI:**
+  - Popup and options UIs are in `src/popup/` and `src/options/`.
+  - Styles are modular and use `common.css` and context-specific CSS.
+
+## Developer Workflows
+- **Build:**
+  - Use `scripts/make.sh [browser]` to build a release ZIP for a specific browser (default: Firefox). This script copies the correct manifest and zips the extension, excluding dev/test files.
+- **Testing:**
+  - Tests are in `src/tests/` and `tests/`. Run them in a browser context; see `CONTRIBUTING.md` for details.
+- **Linting:**
+  - Run ESLint with `eslint` (no AI; `npm run lint` is not yet setup) (configured for JSDoc and modern JS).
```

Comment:
The AI should be able to run eslint if needed.

Reply:
```suggestion
  - Run ESLint with `eslint` (configured for JSDoc and modern JS).
```

Yeah that was my poor wording it had hallucinated `npm run lint` :sweat_smile: